hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseMergeHelper.java

```diff
@@ -36,7 +37,7 @@ public abstract class BaseMergeHelper {
    * @param upsertHandle Merge Handle
    * @throws IOException in case of error
    */
-  public abstract void runMerge(HoodieTable<?, ?, ?, ?> table, HoodieMergeHandle<?, ?, ?, ?> upsertHandle) throws IOException;
+  public abstract void runMerge(HoodieTable<?, ?, ?, ?> table, HoodieMergeHandle<?, ?, ?, ?> upsertHandle,  Option<String[]> partitionFields, Object[] partitionValues) throws IOException;
```

Comment:
This is a core method that runs the merge. I don't think it needs to know about partition fields and values. Is there no way we can get them without adding the arguments? `HoodieMergeHandle` has the file path and we know the base path from `HoodieTable` which has the meta client, so maybe we can get the necessary fields without changing the signature.

Reply:
I agree that we shouldn't add the new params to the method. Would it be ok to add a fields to the merge handle and set those

Reply:
Yeah I think that's fine.